kore/src/Kore/Step/Rule.hs

```diff
@@ -148,15 +155,72 @@ newtype AxiomPatternError = AxiomPatternError ()
 
 instance NFData AxiomPatternError
 
+{-| Defines the right-hand-side of a rewrite rule / claim
+-}
+data RHS variable = RHS
+    { existentials :: ![TermLike.ElementVariable variable]
```

Comment:
Should this be a `Set`?

Reply:
I could be, but I was reluctant to make it a set because then rhsToTerm would not be the inverse of termToRHS.  Do you think we should make it a Set?

Reply:
> then rhsToTerm would not be the inverse of termToRHS

Why not?

Reply:
because when we fold the set to add back the existentials, they might come up in a different orde. So, although semantically equivalent, they won't be syntactically equivalent.  If you think we shouldn't worry about that, I don't mind making it a set.

Reply:
Oh, I see. It's nice to have the two functions be inverses of each other, so let's not use a `Set` then.